### PR TITLE
update packaging pipelines in preparation for the release

### PR DESCRIPTION
**Pull request by askhade**
This Updates
1. Ubuntu to 20.04
2. Cuda to 11.8
3. Torch to 2.0.0
4. ONNXRuntime-Training to latest nightly and stable (built with cu118)
5. Numerous other dependency upgrade like numpy etc..
6. Repo clean-up to remove old Dockerfiles

### CODE REVIEW

File: torch_ort/docker/Dockerfile.onnxruntime-torch2.0.0-cu118-cudnn8-devel-ubuntu20.04 [hunk 1]
```diff
@@ -34,10 +43,13 @@ RUN pip install --upgrade pip
 
 # PyTorch
 RUN pip install onnx ninja
-RUN pip install torch==${TORCH_VERSION}+${TORCH_CUDA_VERSION} torchvision==${TORCHVISION_VERSION} -f https://download.pytorch.org/whl/torch/
+RUN pip install torch==${TORCH_VERSION} torchvision==${TORCHVISION_VERSION} -f https://download.pytorch.org/whl/${TORCH_CUDA_VERSION}
```
Comment (baijumeswani) [inline torch_ort/docker/Dockerfile.onnxruntime-torch2.0.0-cu118-cudnn8-devel-ubuntu20.04:46]: Seems like this is using `/usr/bin/pip`? Not sure if we want to use miniconda or /usr/bin/pip.

I think we could use miniconda over /usr/bin/pip.

Maybe this can happen later though.

File: tools/ci_build/github/linux/docker/manylinux2014_build_scripts/cpython-pubkey-310-311.txt
Comment (baijumeswani) [inline tools/ci_build/github/linux/docker/manylinux2014_build_scripts/cpython-pubkey-310-311.txt:1]: Why is this file required?

File: tools/ci_build/github/linux/docker/manylinux2014_build_scripts/build_openssl.sh
Comment (baijumeswani) [inline tools/ci_build/github/linux/docker/manylinux2014_build_scripts/build_openssl.sh:1]: Why do we need openssl?

**Review by baijumeswani — APPROVED** [review]

File: tools/ci_build/github/linux/docker/manylinux2014_build_scripts/cpython-pubkey-310-311.txt
Comment (askhade) [reply tools/ci_build/github/linux/docker/manylinux2014_build_scripts/cpython-pubkey-310-311.txt:1]: Needed for cpython 3.10 and 3.11 to verify the downloads. Equivalent to cpython-pubkey.txt

File: tools/ci_build/github/linux/docker/manylinux2014_build_scripts/build_openssl.sh
Comment (askhade) [reply tools/ci_build/github/linux/docker/manylinux2014_build_scripts/build_openssl.sh:1]: Needed for building cpython. We copy these scripts from manylinux instead we should add manylinux as a submodule (similar to ort) then we can get rid of the manylinux2014_build_scripts folder.

File: torch_ort/docker/Dockerfile.onnxruntime-torch2.0.0-cu118-cudnn8-devel-ubuntu20.04 (hunk 1 above)
Comment (askhade) [reply torch_ort/docker/Dockerfile.onnxruntime-torch2.0.0-cu118-cudnn8-devel-ubuntu20.04:46]: No, this should be using miniconda. miniconda is higher up in PATH.

Comment (baijumeswani) [reply torch_ort/docker/Dockerfile.onnxruntime-torch2.0.0-cu118-cudnn8-devel-ubuntu20.04:46]: I am not sure, but my guess is that although miniconda is higher up in PATH, the new PATH has not been loaded yet. So, miniconda pip is not in use yet.

Comment (baijumeswani) [reply torch_ort/docker/Dockerfile.onnxruntime-torch2.0.0-cu118-cudnn8-devel-ubuntu20.04:46]: [Here](https://github.com/microsoft/onnxruntime/blob/main/tools/ci_build/github/linux/docker/scripts/install_python_deps.sh#L31-L36) is what we do in `onnxruntime`.

Comment (askhade) [reply torch_ort/docker/Dockerfile.onnxruntime-torch2.0.0-cu118-cudnn8-devel-ubuntu20.04:46]: ok let me test and fix this

**Review by AdamLouly — APPROVED** [review]
LGTM
